src/main/java/org/opensearch/sdk/SDKHttpRequest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.http.HttpRequest;
+import org.opensearch.http.HttpResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class helps to get instance of RestRequest
+ */
+public class SDKHttpRequest implements HttpRequest {
+    private final RestRequest.Method method;
+    private final BytesReference content;
+    private final Map<String, List<String>> headers;
+    private final BytesReference content;
+
+    /**
+    * Instantiates this class with a copy of request
+    *
+    * @param request The request
+    */
+    public SDKHttpRequest(ExtensionRestRequest request) {
+        this.method = request.method();
+        this.content = request.content();
+        this.headers = request.headers();
+        this.content = request.content();
```

Comment:
you've duplicated content and are still missing the uri or path (I'm not sure the difference, you might need to look into that.)